lib/rubocop/cop/rspec/described_class.rb

```diff
@@ -38,27 +52,42 @@ def on_block(node)
           describe, described_class, body = described_constant(node)
           return unless top_level_describe?(describe)
 
-          find_constant_usage(body, described_class) do |match|
-            add_offense(match, :expression, format(MSG, match.const_name))
+          @described_class = described_class
```

Comment:
I usually try to avoid setting instance variables in the `on_*` methods that are used by private methods. I think it makes it a bit easier to reason about if the various cop methods aren't depending on state that may or may not be set. I think part of the reason I prefer this (and it seems to be the style in most regular rubocop cops) is because it is sort of unclear what the lifecycle of a cop is. It seems like it gets initialized once per file but I actually haven't checked to confirm this before :laughing: